Pulled my tv-show-dom-project repo for review.

mikinet opened the pull request.

mikinet committed:
Modified the html page title.

mikinet committed:
Completed Level 100 of the project.
Edited script.js and style.css files to meet requirements of the level.

mikinet committed:
Merge pull request #1 from mikinet/level-100

Completed Level 100 of the project.

mikinet committed:
Completed Level 200 of the project.
Added more code to script.js and style.css files needed for the level.

mikinet committed:
Merge pull request #2 from mikinet/level-200

Completed Level 200 of the project.

mikinet committed:
Completed Level 300 of the project.
Added more code to script.js and style.css files needed for the level.

mikinet committed:
Merge pull request #3 from mikinet/level-300

Completed Level 300 of the project.

mikinet committed:
Refactored code in files (script.js, style.css).
Also changed the behaviour of the <select> and <input> elements.

mikinet committed:
Merge pull request #4 from mikinet/improvements

Refactored code in files (script.js, style.css).

mikinet committed:
Improved the visual and functional features.

mikinet committed:
Merge pull request #6 from mikinet/improvements

Improved the visual and functional features.

mikinet committed:
Improved the code quality.
Deleted styles not used. Done some refactoring.

mikinet committed:
Merge pull request #7 from mikinet/improvements

Improved the code quality.

mikinet committed:
Refactored the code for improved code quality.

mikinet committed:
Merge pull request #8 from mikinet/improvements

Refactored the code for improved code quality.

mikinet committed:
Bad commit.

mikinet committed:
Merge branch 'master' of https://github.com/mikinet/tv-show-dom-project

mikinet committed:
Revert "Refactored the code for improved code quality."

mikinet committed:
Merge pull request #9 from mikinet/revert-8-improvements

Revert "Refactored the code for improved code quality."

mikinet committed:
Merge branch 'master' of https://github.com/mikinet/tv-show-dom-project

mikinet committed:
Completed Level 300 of the project.

mikinet committed:
Refactored codes in files for improved quality.
Also replaced the <button> element by a <span> in script.js.

mikinet committed:
Merge pull request #10 from mikinet/refactored

Refactored codes in files for improved quality.

mikinet committed:
Completed Level 400 of the project.
Updated files to meet the requirements of Level 400 of the project.

mikinet committed:
Merge pull request #11 from mikinet/level-400

Completed Level 400 of the project.

mikinet committed:
Completed Level 500 of the project.

Updated files (script.js, style.css) to meet the requiremnts of the level.

mikinet committed:
Updated file to fulfill the Level 500 rquirements.
Improved the behaviour of the show selector during search.
It can now the title of the first show with the search keyword(s) data.

mikinet committed:
Improved the styling. Still needs improvements.

mikinet committed:
Merge pull request #12 from mikinet/level-500

Completed Level 500 of the project.

mikinet committed:
Edited the file. Removed a styling rule.

Removed a styling rule from a media query: @media screen and (min-width: 769px). This reverts a change made in the last commit (#9135e54).

mikinet committed:
Merge pull request #13 from mikinet/level-500

Edited the file. Removed a styling rule.

mikinet committed:
Edited a few css rules. Deleted some others.
Increased the page's top margin. Edited css rules for show thumbnails.
Also removed apparently ineffectual rules. Displaced a styling rule.

mikinet committed:
Merge branch 'master' into level-500

mikinet committed:
Merge pull request #14 from mikinet/level-500

Edited a few CSS rules. Deleted some others.

cbaggini commented:
# Level   100

1. All episodes must be shown   - Good
2. For each episode, AT LEAST   following must be displayed:
1. the episode's name - Good
2. the season number - Good
3. the episode number - Good
4. the episode's medium-sized   image - Good
5. the episode's summary text   - Good
3. You should combine season   number and episode number into an episode code:
1. Each part should be   zero-padded to two digits. - Good
2. Example: S02E07 would be the code for the 7th episode of the 2nd season. S2E7 would be incorrect. - Good
4. Your page should state   somewhere that the data has (originally) come from TVMaze.com, and link back   to that site (or the specific episode on that site). See   tvmaze.com/api#licensing. - Good
 
# Level 200
 
2. Add a "live"   search input:
1. Only episodes whose summary   OR name contains the search   term should be displayed - Good
2. The search should be case-insensitive - Mostly good - searching is case-sensitive though, if I switch to uppercase halfway through a word
3. The display should update immediately after each keystroke   changes the input. - Good
4. Display how many episodes   match the current search - Good
5. If the search box is   cleared, all episodes should   be shown. - Good
 
 
# Level 300
 
Add a select input which allows you to jump quickly to an episode:
 
1. The select input should   list all episodes in the format: "S01E01 - Winter is Coming"  -Good
2. When the user makes a   selection, they should be taken directly to that episode in the list - Good
3. Bonus: if you prefer, when   the select is used, ONLY show the selected episode. If you do this, be sure   to provide a way for the user to see all episodes again. - The "All episodes" dropdown option is not working
 
# Level 350
When your page loads, it must load the   episodes (for the SAME show) from TVMaze API
Good
 
Your search and episode selector must   continue to work
Good, although the "All episodes" dropdown option is not working
 
Your page MUST NOT re-fetch the episodes   every time the user types a character into your search field
Good
 
# Level 400
Add a select input which allows you to   choose which show you are interested in
Good
 
When a show is selected, your app should   display the episodes for that show
Good
 
Ensure that your search and episode   selector controls still work correctly when you switch shows.
Good
 
This show select must list shows in   alphabetical order, case-insensitive.
You seem to have shows ordered by decreasing rating, was this intentional?
 
# Level 500
When your app starts, present a listing   of all shows
Good
 
For each show, you must display at least   name, image, summary, genres, status, rating, and runtime.
Good
 
When a show name is clicked, your app   should fetch and present episodes from that show
Good
 
hide the "shows listing" view
Good 
 
Add a navigation link to enable the user   to return to the "shows listing"
Good
 
Provide a free-text show search through   show names, genres, and summary texts
Mostly good - searching through summary is case-sensitive though, if I switch to uppercase halfway through a word
 
Ensure that your episode search and   episode selector controls still work correctly when you switch from shows   listing to episodes listing and back
Good
 
# Overall comments

Very good work!
You may want to replace http with https in the links for the show images (lots of warnings in the console and some browsers may actually block it from loading completely).
I could have used a few more comments at the start of the script, going further down the code is well commented. 
I really like how you used your functions for both series and episodes! The only downside is that the functions then become quite long. Especially populateItems and performActions also have a quite complex logic. Maybe worth thinking whether putting them in separate functions would cause you to repeat a lot of code or not.
It may also be worth making sure the dropdown menus have borders, it's not very clear they are even there, especially on a large screen, where the dropdown arrow is very far from the text. Also making the dropdown menu narrower on large screens may help. The search function icon is really good, love t!
There are a couple of bugs (see section above) but overall it works really well and the code is nice and clean.

mikinet committed:
Improved the Level 500 product of the project.
Removed code from script.js. Moved refactored code to other files.

mikinet committed:
Modified the structur of the repo.
Created and files.
Added JavaScript files (common.js, episodes-page.js, shows-page.js).
Refactored script from script.js moved distributed to the new files.

mikinet committed:
Edited style.css, index.html.
Modified some rules (style.css). Added  new script tags (index.html).

mikinet committed:
Merge pull request #15 from mikinet/improvements

Improvements

SenaitEkubai commented on pages/common.js:
```diff
@@ -0,0 +1,182 @@
+}
+
+// attach event listeners to elements
+function createEventListeners(pageName) {
```
I like that you have created this function. makes your code clean and compact

mikinet replied:
Thank you for the encouraging comment @SenaitEkubai. I appreciate it.

SenaitEkubai commented on pages/common.js:
```diff
@@ -0,0 +1,182 @@
+// set initial styling to some page components
+function setInitialStyling(pageName) {
+  const contentDiv = document.querySelector(`.all-${pageName}`);
+  const thumbnails = contentDiv.childNodes;
```
why did you name your variable thumbnails. Just curious

mikinet replied:
It's simply because the containers look a bit like thumbnails when you see them, especially the ones in the episodes page. :-)

SenaitEkubai commented on pages/episodes-page.js:
```diff
@@ -0,0 +1,164 @@
+    // on successful response
+    .then((data) => makePageForEpisodes(data, showsList, showId))
+    // on failed response
+    .catch(() => createInitialContent("error"));
```
I would use cosole.error as this would capture the error a display it in the console

mikinet replied:
Thanks @SenaitEkubai for your feedback.
I put that code in line 18 to turn the page into one that is similar to "Error: 404" in an event the episodes page fails to load. 
The thing is, if I used console.error(), the user wouldn't know an error occurred, unless they open "dev tools". That's what I thought anyway.

mikinet replied:
Hi @cbaggini, 
Thank you for your feedback. It's comprehensive and well organised, I really like it! :-) Thank you for taking the time to review my work, too.
That being said, I have already started making improvements that hopefully address the issues mentioned (and maybe more). Cheers. :-)

mikinet commented:
> # Level 100
> 1. All episodes must be shown   - Good
> 2. For each episode, AT LEAST   following must be displayed:
> 3. the episode's name - Good
> 4. the season number - Good
> 5. the episode number - Good
> 6. the episode's medium-sized   image - Good
> 7. the episode's summary text   - Good
> 8. You should combine season   number and episode number into an episode code:
> 9. Each part should be   zero-padded to two digits. - Good
> 10. Example: S02E07 would be the code for the 7th episode of the 2nd season. S2E7 would be incorrect. - Good
> 11. Your page should state   somewhere that the data has (originally) come from TVMaze.com, and link back   to that site (or the specific episode on that site). See   tvmaze.com/api#licensing. - Good
> 
> # Level 200
> 1. Add a "live"   search input:
> 2. Only episodes whose summary   OR name contains the search   term should be displayed - Good
> 3. The search should be case-insensitive - Mostly good - searching is case-sensitive though, if I switch to uppercase halfway through a word
> 4. The display should update immediately after each keystroke   changes the input. - Good
> 5. Display how many episodes   match the current search - Good
> 6. If the search box is   cleared, all episodes should   be shown. - Good
> 
> # Level 300
> Add a select input which allows you to jump quickly to an episode:
> 
> 1. The select input should   list all episodes in the format: "S01E01 - Winter is Coming"  -Good
> 2. When the user makes a   selection, they should be taken directly to that episode in the list - Good
> 3. Bonus: if you prefer, when   the select is used, ONLY show the selected episode. If you do this, be sure   to provide a way for the user to see all episodes again. - The "All episodes" dropdown option is not working
> 
> # Level 350
> When your page loads, it must load the episodes (for the SAME show) from TVMaze API
> Good
> 
> Your search and episode selector must continue to work
> Good, although the "All episodes" dropdown option is not working
> 
> Your page MUST NOT re-fetch the episodes every time the user types a character into your search field
> Good
> 
> # Level 400
> Add a select input which allows you to choose which show you are interested in
> Good
> 
> When a show is selected, your app should display the episodes for that show
> Good
> 
> Ensure that your search and episode selector controls still work correctly when you switch shows.
> Good
> 
> This show select must list shows in alphabetical order, case-insensitive.
> You seem to have shows ordered by decreasing rating, was this intentional?
> 
> # Level 500
> When your app starts, present a listing of all shows
> Good
> 
> For each show, you must display at least name, image, summary, genres, status, rating, and runtime.
> Good
> 
> When a show name is clicked, your app should fetch and present episodes from that show
> Good
> 
> hide the "shows listing" view
> Good
> 
> Add a navigation link to enable the user to return to the "shows listing"
> Good
> 
> Provide a free-text show search through show names, genres, and summary texts
> Mostly good - searching through summary is case-sensitive though, if I switch to uppercase halfway through a word
> 
> Ensure that your episode search and episode selector controls still work correctly when you switch from shows listing to episodes listing and back
> Good
> 
> # Overall comments
> Very good work!
> You may want to replace http with https in the links for the show images (lots of warnings in the console and some browsers may actually block it from loading completely).
> I could have used a few more comments at the start of the script, going further down the code is well commented.
> I really like how you used your functions for both series and episodes! The only downside is that the functions then become quite long. Especially populateItems and performActions also have a quite complex logic. Maybe worth thinking whether putting them in separate functions would cause you to repeat a lot of code or not.
> It may also be worth making sure the dropdown menus have borders, it's not very clear they are even there, especially on a large screen, where the dropdown arrow is very far from the text. Also making the dropdown menu narrower on large screens may help. The search function icon is really good, love t!
> There are a couple of bugs (see section above) but overall it works really well and the code is nice and clean.

mikinet commented:
Hi @cbaggini, 
Thank you for your feedback. It's comprehensive and well organised, I really like it! :-) Thank you for taking the time to review my work, too.
That being said, I have already started making improvements that hopefully address the issues mentioned (and maybe more). Cheers. :-)

SenaitEkubai commented:
Thanks Miko,
I really appreciate your detailed feedback, I know the css is not complete
yet so I am working on making it responsive.
I will work on implementing your comments  in the coming days.
thanks again for your feedback

On Fri, Mar 5, 2021 at 5:10 PM mikinet <notifications@github.com> wrote:

> Level 100
>
>    1. All episodes must be shown - Good
>    2. For each episode, AT LEAST following must be displayed:
>    3. the episode's name - Good
>    4. the season number - Good
>    5. the episode number - Good
>    6. the episode's medium-sized image - Good
>    7. the episode's summary text - Good
>    8. You should combine season number and episode number into an episode
>    code:
>    9. Each part should be zero-padded to two digits. - Good
>    10. Example: S02E07 would be the code for the 7th episode of the 2nd
>    season. S2E7 would be incorrect. - Good
>    11. Your page should state somewhere that the data has (originally)
>    come from TVMaze.com, and link back to that site (or the specific episode
>    on that site). See tvmaze.com/api#licensing. - Good
>
> Level 200
>
>    1. Add a "live" search input:
>    2. Only episodes whose summary OR name contains the search term should
>    be displayed - Good
>    3. The search should be case-insensitive - Mostly good - searching is
>    case-sensitive though, if I switch to uppercase halfway through a word
>    4. The display should update immediately after each keystroke changes
>    the input. - Good
>    5. Display how many episodes match the current search - Good
>    6. If the search box is cleared, all episodes should be shown. - Good
>
> Level 300
>
> Add a select input which allows you to jump quickly to an episode:
>
>    1. The select input should list all episodes in the format: "S01E01 -
>    Winter is Coming" -Good
>    2. When the user makes a selection, they should be taken directly to
>    that episode in the list - Good
>    3. Bonus: if you prefer, when the select is used, ONLY show the
>    selected episode. If you do this, be sure to provide a way for the user to
>    see all episodes again. - The "All episodes" dropdown option is not working
>
> Level 350
>
> When your page loads, it must load the episodes (for the SAME show) from
> TVMaze API
> Good
>
> Your search and episode selector must continue to work
> Good, although the "All episodes" dropdown option is not working
>
> Your page MUST NOT re-fetch the episodes every time the user types a
> character into your search field
> Good
> Level 400
>
> Add a select input which allows you to choose which show you are
> interested in
> Good
>
> When a show is selected, your app should display the episodes for that show
> Good
>
> Ensure that your search and episode selector controls still work correctly
> when you switch shows.
> Good
>
> This show select must list shows in alphabetical order, case-insensitive.
> You seem to have shows ordered by decreasing rating, was this intentional?
> Level 500
>
> When your app starts, present a listing of all shows
> Good
>
> For each show, you must display at least name, image, summary, genres,
> status, rating, and runtime.
> Good
>
> When a show name is clicked, your app should fetch and present episodes
> from that show
> Good
>
> hide the "shows listing" view
> Good
>
> Add a navigation link to enable the user to return to the "shows listing"
> Good
>
> Provide a free-text show search through show names, genres, and summary
> texts
> Mostly good - searching through summary is case-sensitive though, if I
> switch to uppercase halfway through a word
>
> Ensure that your episode search and episode selector controls still work
> correctly when you switch from shows listing to episodes listing and back
> Good
> Overall comments
>
> Very good work!
> You may want to replace http with https in the links for the show images
> (lots of warnings in the console and some browsers may actually block it
> from loading completely).
> I could have used a few more comments at the start of the script, going
> further down the code is well commented.
> I really like how you used your functions for both series and episodes!
> The only downside is that the functions then become quite long. Especially
> populateItems and performActions also have a quite complex logic. Maybe
> worth thinking whether putting them in separate functions would cause you
> to repeat a lot of code or not.
> It may also be worth making sure the dropdown menus have borders, it's not
> very clear they are even there, especially on a large screen, where the
> dropdown arrow is very far from the text. Also making the dropdown menu
> narrower on large screens may help. The search function icon is really
> good, love t!
> There are a couple of bugs (see section above) but overall it works really
> well and the code is nice and clean.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mikinet/tv-show-dom-project/pull/5#issuecomment-791556033>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARP67Y2YOOCAJIK7BDOTAQDTCEF7PANCNFSM4XX73S5A>
> .
>

mikinet committed:
Implemented shows api call, with added feature.
Api call implemented to fetch shows data directly from TVMaze.com.
Removed files. Added new files with updated, improved script.

mikinet committed:
Updated index.html and style.css.
Updated the files to reflect changes applied to the project.

mikinet committed:
Merge pull request #16 from mikinet/improvements

Improvements

mikinet committed:
Moved a line of code to views/common.js.
Removed code applicable to both shows and episodes pages.

mikinet committed:
Added a line of code to file (line 2).

mikinet committed:
Uncommented temporarily commented out line (14).

mikinet committed:
Merge pull request #17 from mikinet/improvements

Improvements made to the project.

mikinet committed:
Removed a line of code from file.
A constant definition moved to shows-page-services.js.

mikinet committed:
Added a line of code (line 1).
Also changed the request format from HTTP to an HTTPS one.

mikinet committed:
Merge pull request #18 from mikinet/improvements

Improvements

mikinet committed:
Fixed a pagination bug (may need more testing).

mikinet committed:
Added comment (line 49).

mikinet committed:
Edited and displaced a css rule (.link).

mikinet committed:
Merge pull request #20 from mikinet/improvements

Improvements
